File: app/src/main/java/top/someones/cardmatch/core/ImageCache.java
package top.someones.cardmatch.core;

import android.graphics.Bitmap;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class ImageCache {
    private static final Map<String, Bitmap> mBaseCache = new ConcurrentHashMap<>();
    private static final Map<String, Bitmap> mWorkshopCache = new ConcurrentHashMap<>();

    public static Bitmap getCache(String uuid) {
        Bitmap bitmap = mBaseCache.get(uuid);
        if (bitmap != null)
            return bitmap;
        return mWorkshopCache.get(uuid);
    }

    public static void addBaseCache(String uuid, Bitmap bitmap) {
        mBaseCache.put(uuid, bitmap);
    }

    public static void addWorkshopCache(String uuid, Bitmap bitmap) {
        if (uuid == null || bitmap == null)
            return;
        mWorkshopCache.put(uuid, bitmap);
    }

    public static void cleanWorkshopCache() {
        mWorkshopCache.clear();
    }
}

File: app/src/main/java/top/someones/cardmatch/core/ImageCache.kt
package top.someones.cardmatch.core

import android.graphics.Bitmap
import java.util.concurrent.ConcurrentHashMap

object ImageCache {
    private val mBaseCache: MutableMap<String, Bitmap> = ConcurrentHashMap()
    private val mWorkshopCache: MutableMap<String, Bitmap> = ConcurrentHashMap()
    fun getCache(uuid: String): Bitmap? {
        val bitmap = mBaseCache[uuid]
        return bitmap ?: mWorkshopCache[uuid]
    }

    fun addBaseCache(uuid: String, bitmap: Bitmap) {
        mBaseCache[uuid] = bitmap
    }

}